src/core/public/application/scoped_history.test.ts

```diff
@@ -268,8 +268,18 @@ describe('ScopedHistory', () => {
       const gh = createMemoryHistory();
       gh.push('/app/wow');
       const h = new ScopedHistory(gh, '/app/wow');
-      expect(h.createHref({ pathname: '' })).toEqual(`/`);
+      expect(h.createHref({ pathname: '' })).toEqual(`/app/wow`);
```

Comment:
I'd also expect:
```
expect(h.createHref({})).toEqual(`/app/wow`);
```
but looks like returns `''`.

Reply:
`prependBasePath` was returning `undefined` when `location.pathname` was undefined. Changed it to return `basepath` instead in https://github.com/elastic/kibana/pull/62407/commits/460abe1b321826e09bdf08eaa181b4cf0d0b3909